### PR TITLE
Add integration test to check ac cache for remote execution

### PR DESCRIPTION
**Pull request by luluz66**
1. Add a test util function to attach invocation ID to context
2. Add an option to set invocation id when exeucting command in integration
test
3. Add a util function in rbetest.go to get action result for failed
   action
4. Move GetStdoutAndStderr from rbeclient to rbetest. This allow me to
   call it in remote_execution_test.go to check the stdout and stdin for
   the failed action result.

### CODE REVIEW

File: enterprise/server/testutil/testcontext/test_context.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+)
+
+func AttachInvocationIDToContext(ctx context.Context, invocationID string) context.Context {
+	requestMetadata := &repb.RequestMetadata{
+		ToolInvocationId: invocationID,
+	}
+	mdBytes, err := proto.Marshal(requestMetadata)
+	if err != nil {
+		log.Errorf("luluz-Debug: failed to marshal")
```
Comment (tylerwilliams) [inline enterprise/server/testutil/testcontext/test_context.go:20]: lol

**Review by tylerwilliams — APPROVED** [review]
LGTM (one nit)

Maybe give brandon a chance to look this over before you submit?

File: enterprise/server/test/integration/remote_execution/remote_execution_test.go [hunk 2]
```diff
@@ -35,6 +36,68 @@ func TestSimpleCommandWithNonZeroExitCode(t *testing.T) {
...
+	invocationID := "testabc123"
+	cmd := rbe.Execute(&repb.Command{
+		Arguments: []string{"sh", "-c", "echo hello && echo bye >&2 && exit 5"},
+		Platform:  platform,
+	}, &rbetest.ExecuteOpts{
+		InvocationID: invocationID,
+	})
+
+	res := cmd.Wait()
+	assert.Equal(t, 5, res.ExitCode, "exit code should be propagated")
```
Comment (bduffany) [inline enterprise/server/test/integration/remote_execution/remote_execution_test.go:87]: nit: consider adding an assertion just after this line that looking up the ActionResult from the normal AC returns NotFound?

**Review by bduffany — APPROVED** [review]
LGTM modulo 1 nit

File: enterprise/server/test/integration/remote_execution/remote_execution_test.go (hunk 2 above)
Comment (luluz66) [reply enterprise/server/test/integration/remote_execution/remote_execution_test.go:88]: Done

File: enterprise/server/testutil/testcontext/test_context.go (hunk 1 above)
Comment (luluz66) [reply enterprise/server/testutil/testcontext/test_context.go]: oops..Done.

**Review by bduffany — APPROVED** [review]
Nice :)

**Pull request merged**